### PR TITLE
mangohud: update to 0.7.0

### PR DESCRIPTION
**Pull request by OriginCode**
Topic Description
-----------------

Update mangohud, introduce new dependencies required by mangohud.

Package(s) Affected
-------------------

`mangohud, mangohud+32` 0.7.0
`libxnvctrl, libxnvctrl+32` 535.54.03

Security Update?
----------------

No

Build Order
-----------

```
libxnvctrl libxnvctrl+32 mangohud mangohud+32
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: runtime-display/libxnvctrl/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,9 @@
+abinfo "Building libXNVCtrl.so ..."
+cd "$SRCDIR"/src/libXNVCtrl/
+export OUTPUTDIR="out"
```
Comment (MingcongBai) [inline runtime-display/libxnvctrl/autobuild/build]: Why was this necessary?

File: runtime-display/libxnvctrl/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+make
+
+abinfo "Installing libXNVCtrl.so ..."
+cd "$SRCDIR"/src/libXNVCtrl/out/
```
Comment (MingcongBai) [inline runtime-display/libxnvctrl/autobuild/build]: Why?

File: runtime-display/libxnvctrl/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,6 @@
+PKGNAME=libxnvctrl
+PKGSEC=x11
+PKGDEP="x11-lib"
+PKGDES="NVIDIA NV-CONTROL X Extension"
```
Comment (MingcongBai) [inline runtime-display/libxnvctrl/autobuild/defines]: X extension for the NVIDIA NV-CONTROL API

File: runtime-display/libxnvctrl/spec [hunk 4]
```diff
@@ -0,0 +1,3 @@
+VER=535.54.03
```
Comment (MingcongBai) [inline runtime-display/libxnvctrl/spec]: Do we need to sync the versions with that of our `nvidia` package? cc @cthbleachbit

File: runtime-optenv32/libxnvctrl+32/autobuild/build [hunk 5]
```diff
@@ -0,0 +1,12 @@
+abinfo "Building libXNVCtrl.so ..."
+export PATH=/opt/32/bin:$PATH
+export CC=i686-pc-linux-gnu-gcc
+
+cd "$SRCDIR"/src/libXNVCtrl/
+export OUTPUTDIR="out"
```
Comment (MingcongBai) [inline runtime-optenv32/libxnvctrl+32/autobuild/build]: Again, not sure why this would have been needed.

File: runtime-optenv32/libxnvctrl+32/autobuild/defines [hunk 6]
```diff
@@ -0,0 +1,7 @@
+PKGNAME=libxnvctrl+32
+PKGSEC=x11
+PKGDEP="x11-lib+32"
+BUILDDEP="32subsystem"
+PKGDES="NVIDIA NV-CONTROL X Extension (optenv32)"
```
Comment (MingcongBai) [inline runtime-optenv32/libxnvctrl+32/autobuild/defines]: Ditto PKGDES change above.

File: runtime-optenv32/libxnvctrl+32/spec [hunk 7]
```diff
@@ -0,0 +1,3 @@
+VER=535.54.03
```
Comment (MingcongBai) [inline runtime-optenv32/libxnvctrl+32/spec]: Ditto version issues above.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: runtime-display/libxnvctrl/spec (hunk 4 above)
Comment (cthbleachbit) [reply runtime-display/libxnvctrl/spec]: Will have to test whether we need to bind versions

File: app-utils/mangohud/autobuild/defines [hunk 8]
```diff
@@ -1,6 +1,6 @@
 PKGNAME=mangohud
 PKGSEC=utils
-PKGDEP="libglvnd glslang mesa dbus"
+PKGDEP="libglvnd glslang mesa dbus libxnvctrl"
 BUILDDEP="mako"
 BUILDDEP__AMD64="${BUILDDEP} nvidia"
```
Comment (cthbleachbit) [inline app-utils/mangohud/autobuild/defines:7]: Arm64 could also use nvidia.

File: app-utils/mangohud/autobuild/defines [hunk 9]
```diff
@@ -1,6 +1,6 @@
 PKGNAME=mangohud
 PKGSEC=utils
-PKGDEP="libglvnd glslang mesa dbus"
+PKGDEP="libglvnd glslang mesa dbus libxnvctrl"
```
Comment (cthbleachbit) [inline app-utils/mangohud/autobuild/defines]: Nit: Make this PKGRECOM.

File: runtime-optenv32/mangohud+32/autobuild/defines [hunk 10]
```diff
@@ -1,6 +1,6 @@
 PKGNAME=mangohud+32
 PKGSEC=utils
-PKGDEP="libglvnd+32 glslang+32 mesa+32 dbus+32"
+PKGDEP="libglvnd+32 glslang+32 mesa+32 dbus+32 libxnvctrl+32"
```
Comment (cthbleachbit) [inline runtime-optenv32/mangohud+32/autobuild/defines:3]: Nit/Ditto: Make this pkg recom

**Review by cthbleachbit — CHANGES_REQUESTED** [review]

File: runtime-display/libxnvctrl/autobuild/build (hunk 1 above)
Comment (OriginCode) [reply runtime-display/libxnvctrl/autobuild/build]: Instead of building the entire `nvidia-settings`, we can build `libxnvctrl` independently.

**Comment by MingcongBai** [comment]
Blocked by upstream issue. https://github.com/flightlessmango/MangoHud/issues/1152

**Comment by MingcongBai** [comment]
Upstream issue resolved by 99c58794a39bf0f2f95d3f02a6c432b638a8e8a8. Version policy testing pending.

File: runtime-display/libxnvctrl/spec (hunk 4 above)
Comment (OriginCode) [reply runtime-display/libxnvctrl/spec]: Tested 470.63.01, works flawlessly with MangoHUD. I'm confident that the version does not need to keep the same with `nvidia`. Will now upgrade to the latest version [535.113.01](https://github.com/NVIDIA/nvidia-settings/releases/tag/535.113.01).

File: runtime-optenv32/libxnvctrl+32/spec (hunk 7 above)
Comment (OriginCode) [reply runtime-optenv32/libxnvctrl+32/spec]: Resolved above.

**Comment by MingcongBai** [comment]
Version matching not necessary, see response from @OriginCode in the review comments.

**Review by MingcongBai — APPROVED** [review]

**Review by cthbleachbit — APPROVED** [review]
LGTM
